app/views/pages/report.ejs

```diff
@@ -555,4 +737,276 @@
   var obfuscatePath = true;
 </script>
 
+<script>
+  /* Page validation functions. 
+  If validation fails and messages to be displayed should be dispatched in the validation function.
+  These aren't done */
+  const validateDemographicsAndGeography = () => {
+      //validate age
+      var hasAge = document.getElementById("Age").value != "";
+      if (!hasAge) {
+        return [false,"<%= __('Please select a age range') %>"];
+      }
+
+      //validate sex
+      var hasGender = radioSet(document.getElementsByName("gender"));
+      if (!hasGender) {
+        return [false,"<%= __('Please select a Biological gender/Sex') %>"];
+      }
+
+      //validate zip/postal code
+      const isInt = parseInt(document.getElementById("postal-code").value);
+      const correctLength = document.getElementById("postal-code").value.length === "<%= locals.zipPlaceHolder %>".length;
+      if (!isInt || !correctLength) {
+        return [false,"<%= __('Please enter a valid postal code for your country') %>"];
+      }
+
+      return [true,undefined];
+  }
+
+  const validateTestingAndIsolation = () => {
+    //validate in close contact
+    var hasContact = radioSet(document.getElementsByName("been-in-contact-with"));
+    if (!hasContact) {
+      return [false,"<%= __('Please select whether you have been in close contact with someone who tested positive for COVID-19') %>"];
+    }
+
+    //validate tested
+    var hasTested = radioSet(document.getElementsByName("been-tested"));
+    if (!hasTested) {
+      return [false,"<%= __('Please select whether you have been tested for COVID-19') %>"];
+    }
+
+    //validate isolation
+    var hasIsolation = radioSet(document.getElementsByName("isolation-status"));
+    if (!hasIsolation) {
+      return [false,"<%= __('Please select your isolation status') %>"];
+    }
+
+    return [true, undefined];
+  }
+
+  const validateSymptoms = () => {
+    var hasSymptom = radioSet(document.getElementsByClassName("symptom"));
+    if (hasSymptom) {
+      const validDate = Date.parse(document.getElementById("symptom-start").value);
+      if (!validDate) {
+        return [false,"<%= __('Please add a start date for your symptoms') %>"];
```

Comment:
The symptom start date has not been required up until now, so we should maybe remove this validation? Could be hard to remember when the symptoms started.

Reply:
This one is done now. I've removed the validation for symptoms